### ISSUE TITLE
Android build failed

### ISSUE THREAD

**Issue by Rakha112** [margelo/react-native-filament#254]
Hey, first of all, thank you so much for creating this library and bringing the 3D world to React Native. I'm experiencing a build failure on Android after installing react-native-filament. This issue only occurs on Android, while it works perfectly on iOS

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-filament:buildCMakeDebug[armeabi-v7a]'.
> com.android.ide.common.process.ProcessException: ninja: Entering directory `/Users/rakha/Documents/PROJECT-DEV/Mobile-Dev/TestingFilament/node_modules/react-native-filament/android/.cxx/Debug/435wh4w6/armeabi-v7a'
  [1/1] Linking CXX shared library ../../../../build/intermediates/cxx/Debug/435wh4w6/obj/armeabi-v7a/libreact-native-filament.so
  FAILED: ../../../../build/intermediates/cxx/Debug/435wh4w6/obj/armeabi-v7a/libreact-native-filament.so 
```

```
C++ build system [build] failed while executing:
      /Users/rakha/Library/Android/sdk/cmake/3.22.1/bin/ninja \
        -C \
        /Users/rakha/Documents/PROJECT-DEV/Mobile-Dev/TestingFilament/node_modules/react-native-filament/android/.cxx/Debug/435wh4w6/armeabi-v7a \
        react-native-filament
    from /Users/rakha/Documents/PROJECT-DEV/Mobile-Dev/TestingFilament/node_modules/react-native-filament/android
```

Steps to reproduce
1. [Clone this repo](https://github.com/Rakha112/FilamentBuildFailed)
2. Install the package ( npm install )
3. Run the app ( npx expo run:android )

**Comment by hannojg** [comment]
Hm, I would recommend upgrading RN to the latest and worklets to the latest if thats possible for you? I am a 100% certain that it will work on the latest versions

**Comment by kuro-mekami** [comment]
Hey, I’m running into this same issue on Android as well. As soon as react-native-filament is installed into a fresh Expo project (alongside react-native-worklets-core), the Android build fails at the linking stage with several undefined symbols referencing `RNWorklet::JsiWorkletContext`:

```
undefined symbol: RNWorklet::JsiWorkletContext::JsiWorkletContext(...)
undefined symbol: RNWorklet::JsiWorkletContext::invokeOnWorkletThread(...)
undefined symbol: typeinfo for RNWorklet::JsiWorkletContext
``` 

This seemingly points to a missing or unlinked library for the JSI “worklet” context on Android. iOS builds fine, so it appears the iOS toolchain is handling that dependency differently than Android. I also see a bunch of warnings about unused functions (`convertJSUnionToEnum` and others), but those don’t seem to be directly related to the actual linker error.

I’m on the latest versions of React Native, React Native Filament, and React Native Worklets Core, and it still fails when I run `npx expo run:android`. But this was also an issue with Expo 51 and older versions of RNF and RNWC. Clearing the Gradle cache, doing a fresh build, and switching between the new and old React Native architectures haven’t made a difference.

Interestingly, I tried the exact same setup in a bare React Native app, and I can build for Android there without any linking errors. So, it seems specific to Expo’s build process. I’ve compared the Gradle config in my bare RN project to the one that Expo sets up, and even brought over some parts like extra build.gradle settings or CMake flags (without breaking anything Expo-specific), but I still get the same undefined symbol errors on Android.

Given that it works fine in bare RN, it seems that Expo’s modified Gradle/CMake configuration or build process is preventing the `RNWorklet::JsiWorkletContext` symbols from being included. I’ve tried tinkering with a bunch of build settings, but couldn't pinpoint exactly where or why the link is failing.

**Comment by sabuhiteymurov** [comment]
I'm experiencing the same issue.

**Comment by sabuhiteymurov** [comment]
> This seems to be an expo specific issue, are you all using expo?

Yes, with the latest versions: expo: "52.0.25", react-native: "0.76.6", react-native-filament: "1.6.0". 

I'm getting this error when running"expo run:android":
`ld.lld: error: undefined symbol: RNWorklet::JsiWorkletContext::JsiWorkletContext(...)
ld.lld: error: undefined symbol: RNWorklet::JsiWorkletContext::invokeOnWorkletThread(...)
ld.lld: error: undefined symbol: typeinfo for RNWorklet::JsiWorkletContext
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
`

**Comment by mustafakendiguzel** [comment]
Same issue

**Comment by mustafakendiguzel** [comment]
Is there anyone working on this? This issue is critical and blocking our project

**Comment by hannojg** [comment]
Interesting, so I can reproduce the issue when I run `npm run android`, however building the project from android studio seems to work. Can you all build with android studio as well?

**Comment by hannojg** [comment]
There seems to be some ABI mismatch during building. RNWC build for `arm64-v8a`  (which I believe is the correct target for my active device), but the compile step tries to link the library for `armeabi-v7a`

(Just posting my findings as I go)

![Image](https://github.com/user-attachments/assets/a3589ad6-517c-4480-8146-0a3fc2adc5ea)

**Comment by hannojg** [comment]
The issue from the reproduction is fixed in:

- https://github.com/margelo/react-native-filament/releases/tag/v1.6.1

**Comment by kuro-mekami** [comment]
Hey, I just tested the new release, and I can now build successfully for Android! Thank you for the quick fix! :)
